src/pages/invoices/common/components/ClientSelector.tsx

```diff
@@ -51,10 +53,19 @@ export function ClientSelector(props: Props) {
   };
 
   useEffect(() => {
-    resource?.client_id &&
-      clientResolver
-        .find(resource.client_id)
-        .then((client) => setClient(client));
+    if (resource?.client_id) {
+      clearTimeout(resolveClientTimeOut.current);
+
+      const currentTimeout = setTimeout(
+        () =>
+          clientResolver
+            .find(resource.client_id)
+            .then((client) => setClient(client)),
+        25
+      );
+
+      resolveClientTimeOut.current = currentTimeout;
+    }
```

Comment:
@beganovich The reason for the bug is that we do not set entities to an `undefined` value upon unmounting `Creation` components. Consequently, `resource.client_id` remains truthy even on the initial render when we have the `"incorrect"` client_id compared to the `passed` client_id that should be utilized. However, the logic cannot discern which one is correct, leading it to call the API on the first render and set the client regardless of subsequent renders.

There are a few ways to resolve this. The first and most straightforward approach is to set the entities to an undefined value when we `unmount` creation components. For instance, when the creation component for Invoices unmounts, we can simply execute `setInvoice(undefined)`.

I opted against this approach due to the current test coverage. Uncertain of potential issues that this change might introduce, especially given the significance of the entities involved, I chose a more secure alternative.

In this alternative approach, we await the last change of `resource.client_id` and then call the API to resolve the client.

Once we achieve extensive test coverage, we can reconsider and potentially adopt a different, perhaps more preferable, approach to address similar bugs.

I hope you agree with this chosen approach.